src/monitor/group_state_machine.c

```diff
@@ -1256,8 +1256,47 @@ ProceedGroupStateForMSFailover(AutoFailoverNode *activeNode,
 	 * So all the expected candidates did report their LSN, no node is missing.
 	 * Let's see about selecting a candidate for failover now, when we do have
 	 * candidates.
+	 *
+	 * To start the selection process, we require at least number_sync_standbys
+	 * nodes to have reported their LSN and be currently healthy, otherwise we
+	 * won't be able to maintain our guarantees: we would end-up with a node in
+	 * WAIT_PRIMARY state with all the writes blocked for lack of standby
+	 * nodes.
 	 */
-	if (candidateList.candidateCount > 0)
+	char *formationId = activeNode->formationId;
+	AutoFailoverFormation *formation = GetFormation(formationId);
+	int minCandidates = formation->number_sync_standbys + 1;
+
+	/* no candidates is a hard pass */
+	if (candidateList.candidateCount == 0)
+	{
+		return false;
+	}
+
+	/* not enough candidates to promote and then accept writes, pass */
+	else if (candidateList.candidateCount < minCandidates)
```

Comment:
Should only use the count of the sync candidates here.

Reply:
That's a good point. Done now.